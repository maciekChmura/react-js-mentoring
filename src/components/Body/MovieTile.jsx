import React, { Component } from 'react';
import styled from 'styled-components';
import MoviePoster from './MoviePoster';
import MovieInfo from './MovieInfo';
import MovieGenre from './MovieGenre';

const MovieTileWrapper = styled.div`
  width: 200px;
  height: 360px;
  display: flex;
  flex-direction: column;
  margin: 10px;
`;

class MovieTile extends Component {
  render() {
    const { poster_path, title, release_date, genres } = this.props.data;
    return (
      <MovieTileWrapper>
        <MoviePoster posterImage={poster_path} />
        <MovieInfo title={title} release_date={release_date} />
        <MovieGenre genres={genres} />
      </MovieTileWrapper>
    );
  }
}

export default MovieTile;
