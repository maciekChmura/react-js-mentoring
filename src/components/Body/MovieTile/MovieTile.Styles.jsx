import styled from 'styled-components';

const MovieTileWrapper = styled.button`
  width: 200px;
  height: 380px;
  display: flex;
  flex-direction: column;
  margin-top: 10px;
  background-color: #fff;
  border: none;
`;

export default MovieTileWrapper;
