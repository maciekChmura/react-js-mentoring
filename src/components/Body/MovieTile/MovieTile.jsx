import React from 'react';
import { connect } from 'react-redux';
import { selectMovie } from '../../../redux/actions';
import MovieTileWrapper from './MovieTile.Styles';
import MoviePoster from '../MoviePoster/MoviePoster';
import MovieInfo from '../MovieInfo/MovieInfo';
import MovieGenre from '../MovieGenre/MovieGenre';

const MovieTile = ({
  // chyba cos tu jest nei tak z data
  data,
  data: { poster_path, title, release_date, genres, id },
  changePage
}) => (
  <MovieTileWrapper
    onClick={() => {
      console.log(data);
      // changePage(id);
      selectMovie(data);
    }}
  >
    <MoviePoster posterImage={poster_path} />
    <MovieInfo title={title} release_date={release_date} />
    <MovieGenre genres={genres} />
  </MovieTileWrapper>
);

export default connect(
  // mapStateToProps,
  { selectMovie }
)(MovieTile);
