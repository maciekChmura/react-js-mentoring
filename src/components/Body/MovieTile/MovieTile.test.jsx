import React from 'react';
import { shallow } from 'enzyme';

import { MovieTile } from './MovieTile';

describe('MovieTile', () => {
  const data = {
    poster_path:
      'https://image.tmdb.org/t/p/w500/1Ilv6ryHUv6rt9zIsbSEJUmmbEi.jpg',
    release_date: '2013-04-18',
    title: 'Iron Man 3',
    genres: ['Action', 'Adventure', 'Science Fiction']
  };
  it('should render correctly', () => {
    const component = shallow(<MovieTile data={data} />);

    expect(component).toMatchSnapshot();
  });
});
