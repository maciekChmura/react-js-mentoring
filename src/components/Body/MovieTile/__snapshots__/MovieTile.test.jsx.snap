// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTile should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieTile
    data={
      Object {
        "genres": Array [
          "Action",
          "Adventure",
          "Science Fiction",
        ],
        "poster_path": "https://image.tmdb.org/t/p/w500/1Ilv6ryHUv6rt9zIsbSEJUmmbEi.jpg",
        "release_date": "2013-04-18",
        "title": "Iron Man 3",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <MoviePoster
          posterImage="https://image.tmdb.org/t/p/w500/1Ilv6ryHUv6rt9zIsbSEJUmmbEi.jpg"
        />,
        <MovieInfo
          release_date="2013-04-18"
          title="Iron Man 3"
        />,
        <MovieGenre
          genres={
            Array [
              "Action",
              "Adventure",
              "Science Fiction",
            ]
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "posterImage": "https://image.tmdb.org/t/p/w500/1Ilv6ryHUv6rt9zIsbSEJUmmbEi.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "release_date": "2013-04-18",
          "title": "Iron Man 3",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "genres": Array [
            "Action",
            "Adventure",
            "Science Fiction",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "MovieTileStyles__MovieTileWrapper-sc-1cuvjis-0",
        "isStatic": true,
        "rules": Array [
          "width:200px;height:380px;display:flex;flex-direction:column;margin-top:10px;background-color:#fff;border-radius:6px;",
        ],
      },
      "displayName": "MovieTileStyles__MovieTileWrapper",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "MovieTileStyles__MovieTileWrapper-sc-1cuvjis-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <MoviePoster
            posterImage="https://image.tmdb.org/t/p/w500/1Ilv6ryHUv6rt9zIsbSEJUmmbEi.jpg"
          />,
          <MovieInfo
            release_date="2013-04-18"
            title="Iron Man 3"
          />,
          <MovieGenre
            genres={
              Array [
                "Action",
                "Adventure",
                "Science Fiction",
              ]
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "posterImage": "https://image.tmdb.org/t/p/w500/1Ilv6ryHUv6rt9zIsbSEJUmmbEi.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "release_date": "2013-04-18",
            "title": "Iron Man 3",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "genres": Array [
              "Action",
              "Adventure",
              "Science Fiction",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "MovieTileStyles__MovieTileWrapper-sc-1cuvjis-0",
          "isStatic": true,
          "rules": Array [
            "width:200px;height:380px;display:flex;flex-direction:column;margin-top:10px;background-color:#fff;border-radius:6px;",
          ],
        },
        "displayName": "MovieTileStyles__MovieTileWrapper",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "MovieTileStyles__MovieTileWrapper-sc-1cuvjis-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
